### PR TITLE
Fix double event fired by QSpinBox in slow graduated

### LINKED ISSUE #31635: UI: any click on number of classes updown triggers twice
In layer properties, whenever I click on the number of classes updown, it gets changed (slowly recalculates), and repeats the action after first recalculation gets finished. Very annoying on slow coloring modes like Quantiles.

image

### PR DESCRIPTION
**Pull request by elpaso**
... renderer widget

Fixes #31635

For the record of the underlying issue:

https://lists.qt-project.org/pipermail/interest/2013-July/007936.html
https://forum.qt.io/topic/82181/qt-doublespin-box-value-changed-slot-has-called-twice
https://forum.qt.io/topic/96094/qspinbox-value-changed-slot-has-called-twice-on-mouse-click
https://www.qtcentre.org/threads/53709-slot-valueChanged(double)-of-doubleSpinBox-called-twice-if-breakpoint-is-set

### CODE REVIEW

File: src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp [hunk 1]
```diff
@@ -906,6 +910,8 @@ void QgsGraduatedSymbolRendererWidget::symmetryPointEditingFinished( )
 
 void QgsGraduatedSymbolRendererWidget::classifyGraduated()
 {
+
+  QApplication::setOverrideCursor( Qt::WaitCursor );
```
Comment (nyalldawson) [inline src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp]: There's a scoped class in GUI which handles this nicely

Comment (nyalldawson) [reply src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp]: (QgsTemporaryCursorOverride)

**Comment by nyalldawson** [comment]
Gosh, that's pretty... yuck. Oh well. 

I'm guessing QgsDoubleSpinBox should have the same workaround?

File: src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp (hunk 1 above)
Comment (elpaso) [reply src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp]: Thanks! I was pretty sure that there was a scoped helper but I didn't search hard enough.
Do you think we could collect all this little helpers in a dedicated doxy section?

**Comment by elpaso** [comment]
> Gosh, that's pretty... yuck. Oh well.
> 
> I'm guessing QgsDoubleSpinBox should have the same workaround?

I think so, it also inherits from QAbstractSpinBox. Would a mixin base class work for us in this case?

**Comment by nyalldawson** [comment]
> I think so, it also inherits from QAbstractSpinBox. Would a mixin base class work for us in this case?

Not in c++ it won't :P

**Comment by elpaso** [comment]
I'm still eating too much Python these days ;)

**Pull request merged**